factory-containers/publish-compose-apps

```diff
@@ -135,22 +139,55 @@ def publish(factory: str, tag: str, app_name: str) -> str:
     return app + '@sha256:' + sha.decode()
 
 
-def main(factory: str, tag: str):
+def main(factory: str, tag: str, platforms: str, sha: str, app_preload_flag: str, app_dir=None):
     convert_docker_apps()
     apps = {}
-    for p in os.listdir():
-        if p.endswith('.disabled'):
+
+    cur_work_dir = os.path.abspath(os.getcwd())
+    if app_dir:
+        os.chdir(app_dir)
+
+    status('Searching for apps in {}'.format(os.path.abspath(os.getcwd())))
+    for app in os.listdir():
+        if app.endswith('.disabled'):
             continue
-        if os.path.exists(os.path.join(p, 'docker-compose.yml')):
-            status('Publishing compose app for: ' + p)
-            uri = publish(factory, tag, p)
-            apps[p] = {'uri': uri}
+        if os.path.exists(os.path.join(app, 'docker-compose.yml')):
+            status('Publishing compose app for: ' + app)
+            uri = publish(factory, tag, app)
+            apps[app] = {'uri': uri}
 
     # this is a bit of a hack. ota-dockerapp.py needs this to create a target:
     with open('/archive/compose-apps.json', mode='w') as f:
         json.dump(apps, f, indent=2)
 
+    if app_preload_flag == '1':
+        # ------ Dumping container images of all apps -----
+        app_images_dir = '/var/cache/bitbake/app-images/' + sha
+        if os.path.exists(app_images_dir):
+            shutil.rmtree(app_images_dir, ignore_errors=True)
+
+        os.makedirs(app_images_dir)
+
+        status('Dumping container images of the published compose apps to ' + app_images_dir)
+
+        prefixed_platforms = platforms.split(',')
+        archs = []
+        for platform in prefixed_platforms:
+            archs.append(platform.split('/')[1])
+
+        with tempfile.TemporaryDirectory() as tmp_dst_dir:
+            dump_app_images(os.path.abspath(os.getcwd()), archs, tmp_dst_dir)
```

Comment:
i think you can replace `os.path.abspath(os.getcwd())` with the `cur_work_dir` you've added on line 146

Reply:
line #148 can change a cur dir, so we have to get it once more here. cur_work_dir is needed here to make sure that execution environment comes back to the original cur work dir. But, all that stuff can be removed, I basically need it for unit & manual testing, it allows me running this script from ci-scripts directory and points it to my factory's containers dir.

Reply:
this all kind of makes sense. the code might be a little less complex if this test-env specific logic was moved into the the __main__ section, but I'm not sure. this is tested, so i'd say leave it.